pallets/extended-assets/src/lib.rs

```diff
@@ -274,7 +284,27 @@ pub mod pallet {
 
             Self::check_regulated_assets_for_binding(&regulated_asset_id, &who)?;
 
+            // In case the regulated asset is already bound to another SBT, we need to unbind it
+            // by removing it from the previous SBT's regulated assets list.
+            if <RegulatedAssetToSoulboundAsset<T>>::contains_key(regulated_asset_id) {
+                let previous_sbt = Self::regulated_asset_to_sbt(regulated_asset_id);
+                <SoulboundAsset<T>>::mutate(previous_sbt, |metadata| {
+                    if let Some(metadata) = metadata {
+                        metadata.regulated_assets.remove(&regulated_asset_id);
+                    }
+                });
+            }
+
+            // Bind the regulated asset to the SBT and update the SBT's metadata
             <RegulatedAssetToSoulboundAsset<T>>::set(regulated_asset_id, sbt_asset_id);
+            <SoulboundAsset<T>>::mutate(sbt_asset_id, |metadata| {
+                if let Some(metadata) = metadata {
+                    metadata
+                        .regulated_assets
+                        .try_insert(regulated_asset_id)
+                        .ok();
```

Comment:
I think we should return error if something went wrong

Reply:
@vovac12 Handled, please check.